ereuse_devicehub/inventory/views.py

```diff
@@ -261,6 +288,47 @@ def dispatch_request(self, id):
         return flask.render_template(self.template_name, form=form, referrer=request.referrer)
 
 
+class NewActionView(View):
+    methods = ['POST']
+    decorators = [login_required]
+    _form = NewActionForm
+
+    def dispatch_request(self):
+        self.form = self._form()
+
+        next_url = url_for('inventory.devices.devicelist')
+        lot_id = self.form.lot.data
+        if lot_id:
+            next_url = url_for('inventory.devices.lotdevicelist', lot_id=lot_id)
+
+        if self.form.validate_on_submit():
+            self.form.save()
+            return flask.redirect(next_url)
+
+
+class NewAllocateView(NewActionView, DeviceListMix):
+    methods = ['POST']
+    _form = AllocateForm
+
+    def dispatch_request(self, lot_id=None):
+        self.form = self._form()
+
+        next_url = url_for('inventory.devices.devicelist')
+        lot_id = self.form.lot.data
```

Comment:
Remove unused paramenter `lot_id` of `dispatch_request()` because always is overrided on line 317.